Library/Homebrew/extend/pathname.rb

```diff
@@ -293,6 +293,11 @@ def resolved_path_exists?
     (dirname+link).exist?
   end
 
+  def expand_realpath
+    raise unless absolute?
+    exist? ? realpath : parent.expand_realpath+basename
+  end
+
```

Comment:
I'd prefer a private method on the sandbox class that accepts an argument over adding another method to pathname (which becomes public api).